lib/puppet/functions/sqlserver/password.rb

```diff
@@ -0,0 +1,15 @@
+# frozen_string_literal: true
+
+# @summary This function exists for usage of a role password that is a deferred function
+Puppet::Functions.create_function(:'sqlserver::password') do
+  dispatch :password do
+    optional_param 'Any', :pass
```

Comment:
Interested in the types here.

I wonder if we should be more prescriptive? For example, would `String` be a better choice?

Reply:
@chelnak the password is not mandetory field here so restricting with String will lead to make it mandetory. That's why I have added to any type. 
Ref : https://github.com/puppetlabs/puppetlabs-sqlserver/blob/main/manifests/login.pp#L59
Please let me know if we wanted to make it required field.